### ISSUE TITLE
Support PHP 8.1

### ISSUE THREAD

**Issue by datlechin** [flarum/framework#3383]
## Bug Report

**Screenshots**
![](https://i.imgur.com/ZsVCle4.png)

**Environment**
- Flarum version: v1.2.0
- Webserver: nginx
- Hosting environment: MAC OS
- PHP version: 8.1.2
- Browser: Chrome
```sh
Deprecated: Return type of Flarum\Foundation\Config::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/ngoquocdat/Documents/Code/flarum/vendor/flarum/core/src/Foundation/Config.php on line 61

Deprecated: Return type of Flarum\Foundation\Config::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/ngoquocdat/Documents/Code/flarum/vendor/flarum/core/src/Foundation/Config.php on line 56

Deprecated: Return type of Flarum\Foundation\Config::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/ngoquocdat/Documents/Code/flarum/vendor/flarum/core/src/Foundation/Config.php on line 66

Deprecated: Return type of Flarum\Foundation\Config::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/ngoquocdat/Documents/Code/flarum/vendor/flarum/core/src/Foundation/Config.php on line 71
```

**Possible Solution**

**Additional Context**

**Comment by SychO9** [comment]
What needs to be done:
* Fork `tobscure/json-api-php` and `axy/sourcemap`
* Fix php 8.1 deprecations by adding the `#[\ReturnTypeWillChange]` attribute where necessary until we can properly fix typings when we up the php version to 8 in 2.0.
* Set `error_reporting = E_ALL` in our PHP testing action https://github.com/flarum/.github/blob/main/.github/workflows/REUSABLE_backend.yml#L72 to catch warnings in the test suite.

**Comment by chrisdenend** [comment]
How does one suppress Type Declaration warnings in PHP?  Just trying to get Flarum working with certain extensions.
